unit/suites/test_reconnect.py

```diff
@@ -66,6 +68,112 @@ def test_02_wrong_auth(self):
         con.close()
         self.srv.stop()
 
+    def test_03_mesh(self):
+        # Multiple servers are not supported on Windows
+        if os.name == 'nt':
+            return
+
+        # Start two servers
+        self.srv.start()
+        self.srv.admin("box.schema.user.create('test', { password = 'test', if_not_exists = true })")
+        self.srv.admin("box.schema.user.grant('test', 'execute', 'universe')")
+
+        self.srv2 = TarantoolServer()
+        self.srv2.script = 'unit/suites/box.lua'
+        self.srv2.start()
+        self.srv2.admin("box.schema.user.create('test', { password = 'test', if_not_exists = true })")
+        self.srv2.admin("box.schema.user.grant('test', 'execute', 'universe')")
+
+        # get_nodes function contains both servers' addresses
+        get_nodes = " \
+            function get_cluster_nodes() \
+                return { '%s:%d', '%s:%d' } \
+            end" % (self.srv.host, self.srv.args['primary'], self.srv2.host, self.srv2.args['primary'])
+
+        # Create get_nodes function on servers
+        self.srv.admin(get_nodes)
+        self.srv2.admin(get_nodes)
+
+        # Create srv_id function (for testing purposes)
+        self.srv.admin("function srv_id() return 1 end")
+        self.srv2.admin("function srv_id() return 2 end")
+
+        # Create a mesh connection, pass only the first server address
+        con = tarantool.MeshConnection([{
+            'host': self.srv.host, 'port': self.srv.args['primary']}],
+            user='test',
+            password='test',
+            get_nodes_function_name='get_cluster_nodes',
+            connect_now=True)
+
+        # Check we work with the first server
+        resp = con.call('srv_id')
+        self.assertIs(resp.data and resp.data[0] == 1, True)
+
+        # Stop the first server
+        self.srv.stop()
+
+        # Check we work with the second server
+        resp = con.call('srv_id')
+        self.assertIs(resp.data and resp.data[0] == 2, True)
+
+        # Stop the second server
+        self.srv2.stop()
+
```

Comment:
Here we can also check that every request (say, ping) causes an error.

Reply:
Refactored